### PR TITLE
remove most ptr_fun

### PR DESCRIPTION
**Pull request by neheb**
Removed in C++17

### CODE REVIEW

File: src/protocol/handshake_manager.cc [hunk 1]
```diff
@@ -60,15 +60,10 @@ HandshakeManager::erase(Handshake* handshake) {
...
-  bool operator () (const rak::socket_address* sa1, const Handshake* p2) const {
-    return p2->peer_info() != NULL && *sa1 == *rak::socket_address::cast_from(p2->peer_info()->socket_address());
-  }
-};
-
 bool
 HandshakeManager::find(const rak::socket_address& sa) {
-  return std::find_if(base_type::begin(), base_type::end(), std::bind1st(handshake_manager_equal(), &sa)) != base_type::end();
+  auto f = [&sa](const Handshake* p2) { return p2->peer_info() && sa == *rak::socket_address::cast_from(p2->peer_info()->socket_address()); };
+  return std::any_of(base_type::begin(), base_type::end(), f);
```
Comment (rakshasa) [inline src/protocol/handshake_manager.cc:66]: Wrong style, should not use an intermediate function variable.

File: src/protocol/handshake_manager.cc [hunk 2]
```diff
@@ -77,7 +72,7 @@ HandshakeManager::erase_download(DownloadMain* info) {
     return info != h->download();
   });
 
-  std::for_each(split, base_type::end(), std::ptr_fun(&handshake_manager_delete_handshake));
+  std::for_each(split, base_type::end(), &handshake_manager_delete_handshake);
```
Comment (rakshasa) [inline src/protocol/handshake_manager.cc:75]: Use lambda.

File: src/protocol/handshake_manager.cc (hunk 1 above)
Comment (neheb) [reply src/protocol/handshake_manager.cc:66]: It’s pretty long though

Comment (rakshasa) [reply src/protocol/handshake_manager.cc:66]: It should be multiple lines, and is what I find to be the best style for these cases.

File: src/protocol/handshake_manager.cc (hunk 2 above)
Comment (neheb) [reply src/protocol/handshake_manager.cc]: Done.

File: src/protocol/handshake_manager.cc (hunk 1 above)
Comment (neheb) [reply src/protocol/handshake_manager.cc]: Done. clang-format seems happy.

**Pull request merged**